common/modules/farm/contexts/src/ctx_helper.rs

```diff
@@ -88,8 +88,18 @@ pub trait CtxHelper:
 
     #[inline]
     fn commit_changes(&self, context: &GenericContext<Self::Api>) {
-        self.reward_reserve().set(context.get_reward_reserve());
-        self.reward_per_share().set(context.get_reward_per_share());
+        match context.get_reward_reserve() {
+            Some(value) => {
+                self.reward_reserve().set(value);
+            }
+            None => {}
+        }
+        match context.get_reward_per_share() {
+            Some(value) => {
+                self.reward_per_share().set(value);
+            }
+            None => {}
+        }
```

Comment:
Use `if let Some(value) = context.get_reward_reserve()` instead of having empty `None` matches. Clippy probably agrees with me :D